Libraries/LibC/unistd.cpp

```diff
@@ -588,14 +588,14 @@ int reboot()
     __RETURN_WITH_ERRNO(rc, rc, -1);
 }
 
-int mount(const char* source, const char* target, const char* fs_type, int flags)
+int mount(int source_fd, const char* target, const char* fs_type, int flags)
 {
-    if (!source || !target || !fs_type) {
-        errno = EFAULT;
+    if (source_fd <= 0 && (!target || !fs_type)) {
+        errno = EBADF;
```

Comment:
No no no, don't return `EBADF` for null strings either. Just check them separately :slightly_smiling_face:

Reply:
Actually don't check `source_fd` at all, let the kernel handle it and return `EBADF` if it wants a valid fd for that FS.